module/extension/FileWatcher/src/FileWatcher.cpp

```diff
@@ -27,142 +29,205 @@ namespace extension {
     using ::extension::FileWatchRequest;
     using Unbind = NUClear::dsl::operation::Unbind<FileWatch>;
 
-    void fswatchCallback(const std::vector<fsw::event>& events, void* r) {
-        FileWatcher& reactor = *reinterpret_cast<FileWatcher*>(r);
+    void FileWatcher::file_watch_callback(uv_fs_event_t* handle, const char* filename, int events, int status) {
 
-        // Look through our filesystem events
-        for (auto& event : events) {
+        // Regain our reactor
+        FileWatcher& reactor = *reinterpret_cast<FileWatcher*>(handle->data);
 
-            // Get our path
-            auto path = event.get_path();
+        // Lock our mutex as we are editing our datastructure
+        std::lock_guard<std::mutex> lock(reactor.paths_mutex);
 
-            // Split our path into events
-            std::string dir;
-            std::string filename;
-            std::tie(dir, filename) = utility::file::pathSplit(path);
-
-            // See if we can find this directory
-            auto handler = reactor.handlers.find(dir);
-            if (reactor.handlers.find(dir) != reactor.handlers.end()) {
-                auto reactions = handler->second.find(filename);
-
-                if (reactions != handler->second.end()) {
-                    for (auto& reaction : reactions->second) {
-                        uint flags = 0;
-                        for (auto& flag : event.get_flags()) {
-                            flags |= flag;
-                        }
+        // Work out what path we are watching
+        std::array<char, 512> pathbuff;
+        pathbuff.fill('\0');
```

Comment:
`std::array<char, 512> pathbuff = {0};`?

Reply:
I don't think that actually works in c++